manifests/plugin/remote_execution/ssh/keys.pp

```diff
@@ -0,0 +1,51 @@
+# = Foreman Proxy Remote Execution SSH plugin key management
+#
+# This class generates and installs Remote Execution SSH keys for Foreman proxy
+#
+# === Parameters:
+#
+# $install_key::        Automatically install generated SSH key to root authorized keys
+#                       which allows managing this host through Remote Execution
+#
+# $ssh_identity_dir::   Directory where SSH keys are stored
+#
+# $ssh_identity_file::  Provide an alternative name for the SSH keys
+#
+# $ssh_keygen::         Location of the ssh-keygen binary
+#
+class foreman_proxy::plugin::remote_execution::ssh::keys (
+  Boolean $install_key = false,
+  Stdlib::Absolutepath $ssh_identity_dir = '/var/lib/foreman-proxy/ssh',
+  String $ssh_identity_file = 'id_rsa_foreman_proxy',
+  String $ssh_keygen = '/usr/bin/ssh-keygen',
+) {
+  $ssh_identity_path = "${ssh_identity_dir}/${ssh_identity_file}"
+
+  file { $ssh_identity_dir:
+    ensure => directory,
+    owner  => $foreman_proxy::user,
+    group  => $foreman_proxy::user,
```

Comment:
I would also make these class parameters. That way you don't need to `include foreman_proxy` to test this.

Reply:
How does this look?